.clang-format

```diff
@@ -0,0 +1,29 @@
+Language: Cpp
+BasedOnStyle: LLVM
+AlignEscapedNewlinesLeft: true
+AlwaysBreakAfterReturnType: TopLevelDefinitions
+BreakBeforeBinaryOperators: All
+BreakBeforeBraces: GNU
+ColumnLimit: 150
+ContinuationIndentWidth: 2
+ForEachMacros: [FOR_EACH_TAIL, FOR_EACH_TAIL_SAFE]
+IndentWidth: 4
+IncludeCategories:
+  - Regex: '^<config\.h>$'
+    Priority: -1
+  - Regex: '^<'
+    Priority: 1
+  - Regex: '^"lisp\.h"$'
```

Comment:
What is the purpose of this?

Reply:
https://clang.llvm.org/docs/ClangFormatStyleOptions.html indicates that this is used for determining the order of `#includes`.  Negative one indicates that these go first. 

I am not sure where the lisp regexp came from. In the next commit, I'll remove this.